daemon/containerd/image.go

```diff
@@ -157,17 +157,26 @@ func (i *ImageService) GetImageManifest(ctx context.Context, refOrID string, opt
 
 	cs := i.client.ContentStore()
 
-	desc, err := i.resolveDescriptor(ctx, refOrID)
+	image, err := i.resolveImage(ctx, refOrID)
 	if err != nil {
 		return nil, err
 	}
 
+	desc := image.Target
 	if containerdimages.IsManifestType(desc.MediaType) {
+		plat := desc.Platform
+		if plat == nil {
+			config, err := image.Config(ctx, cs, platform)
```

Comment:
For a follow-up, I wonder if  some of this needs to be combined with `resolveDescriptor` and/or `resolveImage`.  I had a very quick glance, and it looks like in some cases we already use those, and directly after do some platform matching; so perhaps having a `platform` argument that can be passed could be useful.

Reply:
This is actually wrong, once we get the descriptor of the image config we need to `readConfig` to get the real config

Reply:
We definitely need a nicer API to get the bits we usually need for sure